sqlx-postgres/src/row.rs

```diff
@@ -48,3 +53,105 @@ impl ColumnIndex<PgRow> for &'_ str {
+            add_debug_entry(&mut debug_map, self, index, column);
+        }
+
+        debug_map.finish()
+    }
+}
+
+macro_rules! debug_types {
+    { $($enum:ident::$variant:ident => $type:ty),* } => {
+        fn add_debug_entry(
+            debug_map: &mut DebugMap<'_, '_>,
+            row: &PgRow,
+            index: usize,
+            column: &PgColumn) {
+                let name = &column.name;
+                match row.try_get_raw(index) {
+                    Ok(value) => {
+                        match column.type_info.0 {
+                            $(
+                                $enum::$variant => add_decoded_entry::<$type>(debug_map, value, name),
+                            )*
+                            _ => add_raw_entry(debug_map, value, name)
+                        }
+                    }
+                    _ => {
+                        debug_map.entry(name, &"NOT FOUND");
+                    }
+                }
+
+        }
+    }
+}
+
+fn add_decoded_entry<'r, T: Decode<'r, Postgres> + Debug>(
+    debug_map: &mut DebugMap<'_, '_>,
+    value: PgValueRef<'r>,
+    name: &UStr,
+) {
+    match T::decode(value.clone()) {
+        Ok(decoded_value) => {
+            debug_map.entry(name, &decoded_value);
+        }
+        _ => {
+            add_raw_entry(debug_map, value, name);
+        }
+    };
+}
+
+fn add_raw_entry(debug_map: &mut DebugMap<'_, '_>, value: PgValueRef, name: &UStr) {
+    match value.format {
+        PgValueFormat::Text => debug_map.entry(name, &value.as_str().unwrap_or_default()),
+        PgValueFormat::Binary => debug_map.entry(name, &value.as_bytes().unwrap_or_default()),
+    };
+}
+
+debug_types! {
+    PgType::Money => PgMoney,
+    PgType::MoneyArray => Vec<PgMoney>,
+    PgType::Void => (),
+    PgType::Bool => bool,
+    PgType::BoolArray => Vec<bool>,
+    PgType::Float4 => f32,
+    PgType::Float4Array => Vec<f32>,
+    PgType::Float8 => f64,
+    PgType::Int4Range => PgRange<i32>,
+    PgType::Int8Range => PgRange<i64>,
+    PgType::Text => String,
+    PgType::TextArray => Vec<String>,
+    PgType::Bpchar => String,
+    PgType::BpcharArray => Vec<String>,
+    PgType::Name => String,
+    PgType::NameArray => Vec<String>,
+    PgType::Varchar => String,
+    PgType::VarcharArray => Vec<String>,
+    PgType::Interval => PgInterval,
+    PgType::IntervalArray => Vec<PgInterval>,
+    PgType::Oid => Oid,
+    PgType::OidArray => Vec<Oid>,
+    PgType::Char => i8,
+    PgType::CharArray => Vec<i8>,
+    PgType::Int2 => i16,
+    PgType::Int2Array => Vec<i16>,
+    PgType::Int4 => i32,
+    PgType::Int4Array => Vec<i32>,
+    PgType::Int8 => i64,
+    PgType::Int8Array => Vec<i64>,
+    PgType::Timestamp => i64,
+    PgType::TimestampArray => Vec<i64>,
+    PgType::Time=> i64,
+    PgType::TimeArray => Vec<i64>,
+    PgType::Timestamptz => i64,
+    PgType::TimestamptzArray => Vec<i64>,
+    PgType::Date => i32,
+    PgType::DateArray => Vec<i32>
```

Comment:
This is duplicating code that already exists in the macros: https://github.com/launchbadge/sqlx/blob/main/sqlx-macros-core/src/database/postgres.rs#L5

I would prefer to just hoist that instead.

Reply:
I was looking for something like this. Thank you.

Reply:
@abonander Is it possible to take advantage of https://github.com/launchbadge/sqlx/blob/main/sqlx-macros-core/src/database/postgres.rs#L5 without making `sqlx-macros-core` a dependency of `sqlx-postgres`?

Reply:
#2970 

@g-bartoszek are you coming back to this PR at some point?

Reply:
If you were waiting for it to be merged, I'm going to do that today.

Reply:
> #2970 
> 
> @g-bartoszek are you coming back to this PR at some point?

Yes, I plan to do it soon.

Reply:
@abonander I've updated this branch to use your changes.